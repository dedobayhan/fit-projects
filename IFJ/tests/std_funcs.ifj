void main()
var int i,
	double d,
	string s;
{
	// int2str
	printstring(int2str(0)+" ");
	printstring(int2str(0-123)+" ");
	printstring(int2str(123)+" ");
	printstring(int2str(0-1234567890)+" ");
	printstring(int2str(1234567890)+" ");
	printstring(int2str(0-2147483647)+" ");
	printstring(int2str(2147483647)+"\n");

	// double2str
	printstring(double2str(0.0)+" ");
	printstring(double2str(0.0-123.0)+" ");
	printstring(double2str(123.0)+" ");
	printstring(double2str(0-1234567890.0)+" ");
	printstring(double2str(1234567890.0)+" ");
	printstring(double2str(0-2147483647.0)+" ");
	printstring(double2str(2147483647.0)+"\n");

	// str2double
	// printdouble
	printdouble(str2double("0.0")); printstring(" ");
	printdouble(str2double("-444.444")); printstring(" ");
	printdouble(str2double("666.666")); printstring(" ");
	printdouble(str2double("1e58")); printstring(" ");
	printdouble(str2double("-1e67")); printstring(" ");
	printdouble(str2double("1.2222e-234")); printstring(" ");
	printdouble(str2double("1.3333e-289")); printstring("\n");
	// chyba
	// printdouble(str2double("1.3333e-500")); printstring("\n");

	// double2int
	// printint
	printint(double2int(0.0)); printstring(" ");
	printint(double2int(0-444.444)); printstring(" ");
	printint(double2int(666.666)); printstring(" ");
	printint(double2int(0-1.2e-58)); printstring("\n");
	// chyba
	// printint(double2int(2147483648.0)); printstring("\n");
	// printint(double2int(0-1.3e+200)); printstring("\n");
	// printint(double2int(1.2222e234)); printstring("\n");
	// printstring("\n");

	// printstring
	printstring("abcdefghijklmnopqrstuvwxyz"+" ");
	printstring("cukrová vata\\candyfloss je \"mňam\""+" ");
	printstring("$$$1$2$3$4$5$$$"+" ");
	printstring(".    \$!@#$%^&*()_+{}:\"<>?|"+" ");
	printstring("abcd\$efgh\\$"+" ");
	printstring("\n");

	// readint
	i := readint();
	printint(i); printstring("\n");

	// readdouble
	d := readdouble();
	printdouble(d); printstring("\n");

	// readstring
	s := readstring();
	printstring(s); printstring("\n");

	// strlen
	printstring("||"+" je "+int2str(strlen(""))+"\n");
	printstring("|a|"+" je "+int2str(strlen("a"))+"\n");
	printstring("|abcdefgh|"+" je "+int2str(strlen("abcdefgh"))+"\n");
	printstring("|abcdefghijklmnopqrstuvwxyzzzzzzzzzzzzzzzzzzzzzzzzz|"+
		" je "+int2str(strlen("abcdefghijklmnopqrstuvwxyzzzzzzzzzzzzzzzzzzzzzzzzz"))+"\n");
	printstring("|\"\$\\$\"|"+" je "+int2str(strlen("\"\$\\$\""))+"\n");

	// sort
	printstring(sort("a$b$c$d$e$f$g$h$i$j$k$l$m$n$o$p$q$r$s$t$u$v$w$x$y$z$")+"\n");
	printstring(sort("aefghtsdfsd$bsdfsdfs$csdfsdfew$dsdfsf$ewerwersdfs$fsdfsdfsdfwewef$gfsfsfsdfwe$hwfwefwf$iwwwwwwwww$jfwfwefw$kfwefwfw$ffwfwfwl$m$n$o$p$q$r$s$t$u$v$w$x$y$z$")+"\n");
	printstring(sort("zzzz$xxxx$yyyy$wwww$rrrr$ttt$ssss$")+"\n");
	printstring(sort("\$zzzz$xxxx$yy\$\$yy$wwww$rrrr$ttt$ssss$\$")+"\n");
	printstring(sort("Lukas Marsik $xzemek02@stud.fit.vutbr.cz $xpolca03@stud.fit.vutbr.cz $Martin Tomec $Boris Prochazka $xmarsi03@stud.fit.vutbr.cz $xproch63@stud.fit.vutbr.cz $xtomec05@stud.fit.vutbr.cz $Petr Zemek $Libor Polcak ") + "\n");
	// chyba
	//
	// printstring("\n");
}
