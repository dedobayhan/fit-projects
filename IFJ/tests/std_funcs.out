0 -123 123 -1234567890 1234567890 -2147483647 2147483647
0 -123 123 -1.23457e+09 1.23457e+09 -2.14748e+09 2.14748e+09
0 -444.444 666.666 1e+58 -1e+67 1.2222e-234 1.3333e-289
0 -444 666 0
abcdefghijklmnopqrstuvwxyz cukrová vata\candyfloss je "mňam" 12345 .    $!@#%^&*()_+{}:"<>?| abcd$efgh\ 
1
1
aaaa
|| je 0
|a| je 1
|abcdefgh| je 8
|abcdefghijklmnopqrstuvwxyzzzzzzzzzzzzzzzzzzzzzzzzz| je 50
|"$\"| je 4
abcdefghijklmnopqrstuvwxyz
aefghtsdfsdbsdfsdfscsdfsdfewdsdfsfewerwersdfsffwfwfwlfsdfsdfsdfwewefgfsfsfsdfwehwfwefwfiwwwwwwwwwjfwfwefwkfwefwfwmnopqrstuvwxyz
rrrrsssstttwwwwxxxxyyyyzzzz
$$zzzzrrrrsssstttwwwwxxxxyy$$yy
Boris Prochazka Libor Polcak Lukas Marsik Martin Tomec Petr Zemek xmarsi03@stud.fit.vutbr.cz xpolca03@stud.fit.vutbr.cz xproch63@stud.fit.vutbr.cz xtomec05@stud.fit.vutbr.cz xzemek02@stud.fit.vutbr.cz 
